### DISCUSSION TITLE
Hide output devices

### DISCUSSION THREAD

**Discussion by FearNaBoinne** [File-New-Project/EarTrumpet#895] [Ideas]
It would be nice if I could select which devices (not) to show, so I can hide extra devices from EarTrumpet's main screen and right-click menu without deactivating them.

Two examples:
1) Soundblaster with SPDIF output... Soundblaster can 'mirror' the sound over the SPDIF in Dolby Surround encodings, but that requires using the analog device, and NOT the Digital Output device, however you can NOT disable the device in sound settings, or output will no longer work over SPDIF
2) Voicemeeter - creates a virtual devices to allow you to mix/modify/filter the sound before it comes out of the desired output. In this case you want just the Voicemeeter virtual device, not the actual output devices, but again, you cannot deactivate it/them in soundsettings, because Voicemeeter still needs them to actually output sound.

Now combine the above two and you have even more undesired devices you really do not want to see in Eartrumpet...

**Comment by Noiden** [comment]
This would be really nice!

**Comment by karasuhebi** [comment]
It looks like it's being discussed over here: https://github.com/File-New-Project/EarTrumpet/discussions/713

**Reply by NiyaShy** [reply]
Not quite. The discussion you linked is about hiding unwanted/duplicate programs/apps, this one is about hiding unneeded audio devices.

**Comment by NiyaShy** [comment]
Can second this one that it would be really nice if you could hide (or at least collapse) unneeded devices. As it currently stands the only way to hide an entry is to disable it via the windows sound settings window, but that might not be an option for some devices.
To add another example, I currently have 3 active audio devices: 
- one of my monitors (only used as a notification sounds output for a single program, so not needed 99% of the time)
- a virtual sound device created by VB-Audio virtual cable
- my actual speakers (only directly used by a selected few programs, the rest is routed through the VB-A device)

Can't disable any of them since they are all used by some program(s), so it would be nice if I could completely hide the monitor entry and collapse my speakers.

**Comment by Tencryn** [comment]
I have many virtual audio devices that I do not need to see. I do **not** need to hide them from the right-click menu as this takes up far less space plus there are some that I need there, but not in the main menu.

Example: I stream Game Capture HD on Discord which outputs its own game audio. This comes with a delay from my Elgato, so I set the audio output device to be a virtual audio cable, the volume stays at 100 and never needs to be changed. This allows the audio to be captured by Discord, but doesn't get sent to my ears. For whatever reason, upon restart, GCHD ignores the output device until I re-set it. Having this specific VAC to be visible in the right click menu is desirable, but it need not be visible in the main menu.

As a proposal for the main menu, having an eye icon that you can toggle would be beneficial. You can then have a button that'll show/hide hidden devices. I'm unsure how such a button would translate to the right-click menu though, but maybe winners can't be choosers. If you want it visible for one menu, it's visible for the other.

Relating to the audio devices in the main menu, I'm not a fan of the scrollbar that appears. I will accidentally change the volume of devices as I scroll, and that's why I would like to hide some devices.

**Comment by benjamin051000** [comment]
Came here to mention this. 

Steam Streaming Microphone and Steam Streaming Speakers are devices that Steam uses to stream gameplay audio from a host computer to a guest computer, but they aren't devices you would ever want to manually select. They are automatically selected/deselected by Steam when you begin streaming a game to another pc, so they should not appear in EarTrumpet. 

All we would need is a settings entry where we could whitelist/blacklist audio devices from eartrumpet.

**Comment by DaEkstrim** [comment]
I have provided a suggestion and working POC here - https://github.com/File-New-Project/EarTrumpet/discussions/713#discussioncomment-2886034

**Comment by alphaqu** [comment]
Hey! For anyone wanting this feature, I have updated a fork (https://github.com/nnyj/EarTrumpet) with this feature to the latest version of EarTrumpet, and published a [release here](https://github.com/alphaqu/EarTrumpet/releases/tag/2.3.0.0%2Bnnyj). 

Have a nice day!

**Reply by sebascontre** [reply]
Hi, love the feature to hide unwanted apps, but this thread is for hide unwanted **devices**, and can't find how to do that in your fork

**Comment by AutonomousCat** [comment]
Looking forward to this.

**Comment by Tzak03** [comment]
Estaba buscnado esconder una salida de audio, pero no hay esa opcion aun
Cuando hay muchas saldias de audio aumenta mucho la barra de sonidos, donde aparecen las 7 salidas de audioy con las apps cada una, la unica forma que encontre para que esto no pase es ir a sonidos de windows y desactivar estas salidas (dispositivos de reproduccion) y ahi ya no aparecen en el eartrumpet
<img width="369" height="1012" alt="image" src="https://github.com/user-attachments/assets/45d9808f-545d-4918-9053-986e9cba0a85" />
<img width="415" height="464" alt="image" src="https://github.com/user-attachments/assets/d577b157-1669-4ddd-8a70-32ef1e9cbad9" />
<img width="365" height="749" alt="image" src="https://github.com/user-attachments/assets/97714f8d-ad93-4da4-82e8-4d449e440811" />
<img width="411" height="464" alt="image" src="https://github.com/user-attachments/assets/4d520ea5-762d-411f-b5ca-e1ad035cca19" />
